### PR TITLE
TS API Update - Get_Connection_Parameters to support WAITING_PROCESSE…

### PR DESCRIPTION
**Pull request by oschwaldp-oci**
…S_OR_MESSAGES is number of unread samples for datareader, 0 for datawriter.  Will return NOT_AVAILABLE until a Receive_Message or Register_Callback operation has happened to give message type knowledge to the receiver.  Add test for Get_Connection_Parameters functionality

### CODE REVIEW

File: dds/FACE/FaceTSS.cpp [hunk 1]
```diff
@@ -193,18 +193,30 @@ void Get_Connection_Parameters(CONNECTION_NAME_TYPE& connection_name,
...
+        if (OpenDDS::DCPS::DCPS_debug_level > 3) {
+          ACE_DEBUG((LM_DEBUG, "Get_Connection_Parameters: returning NOT_AVAILABLE due to receiver's status not valid\n"));
+        }
         return_code = NOT_AVAILABLE;
         return;
       }
       cur_status.LAST_MSG_VALIDITY = receiver.last_msg_header.message_validity;
-      if (entities.receivers_[connection_id].total_msgs_recvd != 0) {
-        cur_status.REFRESH_PERIOD = entities.receivers_[connection_id].sum_recvd_msgs_latency/entities.receivers_[connection_id].total_msgs_recvd;
+      if (entities.receivers_[connection_id]->total_msgs_recvd != 0) {
```
Comment (iamtheschmitzer) [inline dds/FACE/FaceTSS.cpp]: Use receiver reference from above here, and below.

File: tests/FACE/GetConnectionParameters/Subscriber/Subscriber.cpp [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    if (status != FACE::RC_NO_ERROR) {
+      std::cout << "ERROR: Get_Connection_Parameters after registering callback returned " << status << std::endl;
+      return static_cast<int>(status);
+    }
+    if (connectionStatus.CONNECTION_DIRECTION != FACE::DESTINATION ||
+        connectionStatus.LAST_MSG_VALIDITY != FACE::INVALID ||
+        connectionStatus.MESSAGE != 1 ||
+        connectionStatus.MAX_MESSAGE != ACE_INT32_MAX ||
+        connectionStatus.MAX_MESSAGE_SIZE != 300 ||
+        connectionStatus.WAITING_PROCESSES_OR_MESSAGES != 0 ||
```
Comment (iamtheschmitzer) [inline tests/FACE/GetConnectionParameters/Subscriber/Subscriber.cpp]: We need one test where WAITING_PROCESSES_OR_MESSAGES  is expected to be non-zero also.
